test/e2e/test/elasticsearch/checks_data.go

```diff
@@ -67,7 +68,7 @@ func (dc *DataIntegrityCheck) Init() error {
 	if err != nil {
 		return err
 	}
-	resp, err := dc.client.Request(context.Background(), indexCreation)
+	resp, err := esClient.Request(context.Background(), indexCreation)
 	defer resp.Body.Close() // nolint
```

Comment:
I need to investigate more but I had this error while running the e2e tests _(maybe it comes from my setup)_:

```
  === RUN   TestMutationHTTPToHTTPS/Add_some_data_to_the_cluster_before_starting_the_mutation
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1069576]

goroutine 340 [running]:
testing.tRunner.func1(0xc00037a500)
	/usr/local/go/src/testing/testing.go:792 +0x387
panic(0x119f7a0, 0x1f21720)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch.(*DataIntegrityCheck).Init(0xc0007261c0, 0x0, 0x0)
	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch/checks_data.go:72 +0x2f6
github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch.Builder.MutationTestSteps.func1(0xc00037a500)
	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch/steps_mutation.go:47 +0x98
testing.tRunner(0xc00037a500, 0xc0000eb720)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
```

I think `defer resp.Body.Close()` should be done after the error check anyway.